runtime/src/main/scala/akka/grpc/AkkaGrpcException.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2009-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.grpc
+
+import com.google.protobuf.any.Any
+import com.google.rpc.{ Code, Status }
+import io.grpc.StatusRuntimeException
+import io.grpc.protobuf.StatusProto
+import collection.JavaConverters._
+
+object AkkaGrpcException {
```

Comment:
It's a bit weird that we now have both a `AkkaGrpcException` and a `GrpcServiceException`.

I wonder if we could unify them, but that might not be easy. For the client side, I think it is nice that `AkkaGrpcException` extends `StatusRuntimeException`, however I'm not sure if it'd make sense to have `GrpcServiceException` extend `StatusRuntimeException`. Needs further investigation.

Reply:
If we keep both I think it would be good with a name on this new one that more clearly expresses what it is.

Reply:
I am looking into your suggestions next 👍

Reply:
I managed to unify them. The only way to make this compatible to the "manual" rich error approach is to `GrpcServiceException` extend `StatusRuntimeException` capturing the raw `io.grpc.Metadata`.